### PR TITLE
Mining Tweaks and Occult Tank

### PR DESCRIPTION
**Pull request by MrLostman**
## About The Pull Request
**Replaces cursed oxygen tank with occult tank**
cursed_tank
Has a single random infinite gas at a random temperature.
```
gas tiers
10% chance to hit tier 1 : nitrogen, carbon_dioxide, nitrous_oxide
40% chance to hit tier 2 : oxygen has fixed room temp
40% chance to hit tier 3 : plasma, nitryl, bz, miasma, pluoxium
10% chance to hit tier 4 : tritium, stimulum, hypernoblium

heat tiers
30% chance to hit tier 1 : cold 0-280
30% chance to hit tier 2 : room temp
30% chance to hit tier 3 : hot 300-1000
10% chance to hit tier 4 : S U P E R   H O T 1001-100000

two 1 to 100 rolls are done to pick the gas and temp then a single gas is picked from the tier.
```
**Cleans up some un-modular code from techwebs and necro chest loot**
**Removes tech disks from loot list adds them to a new node Experimental Mining Technology**

## Why It's Good For The Game
Flab loot is lazy coding/game design and should be purged. The meme tank tends to be useful and boring but can be a bit of a joke more interesting objects the better. Simple to nerf if it becomes a problem.

## Changelog
:cl:
add: added occult tank
del: removed cursed oxygen tank
tweak: removed hyperaoemod, repeatermod, resonatormod and bountymod tech disks from necro chest loot list
tweak: adds hyperaoemod, repeatermod, resonatormod and bountymod to advanced mining
/:cl:

### CODE REVIEW

**Comment by Mick1299** [comment]
Advanced mining tech looks a little bloated now, maybe add a new tech node that branches off mining tech

**Comment by MrLostman** [comment]
its not really outside the general size variation of tech nodes, but I don't really mind making a new node

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -592,7 +592,7 @@
 	display_name = "Advanced Mining Technology"
 	description = "Efficiency Level 127"	//dumb mc references
 	prereq_ids = list("basic_mining", "adv_engi", "adv_power", "adv_plasma")
-	design_ids = list("drill_diamond", "jackhammer", "hypermod", "plasmacutter_adv", "mech_kinetic_accelerator") //austation — MKA 27
+	design_ids = list("drill_diamond", "jackhammer", "hypermod", "plasmacutter_adv")
```
Comment (TheFakeElon) [inline code/modules/research/techweb/all_nodes.dm:595]: you can modularise this

**Comment by TheFakeElon** [comment]
finally, someone else actually modularising their techweb changes.

**Comment by jbox144** [comment]
this change is Compilatron Approved:tm:

**Comment by TheFakeElon** [comment]
> this change is Compilatron Approved™️

this provide little value

File: austation/code/modules/mining/necropolis_chests.dm [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      new /obj/item/borg/upgrade/modkit/lifesteal(src)
+      new /obj/item/bedsheet/cult(src)
+    if(26)
+      new /obj/item/clothing/neck/necklace/memento_mori(src)
+    if(27)
+      new /obj/item/reagent_containers/glass/waterbottle/relic(src)
+    if(28)
+      new /obj/item/reagent_containers/glass/bottle/necropolis_seed(src)
+    if(29)
+      new /obj/item/tank/internals/occult(src)
```
Comment (MCterra10) [inline austation/code/modules/mining/necropolis_chests.dm:63]: ok this is cool, but if bee adds a new tendril item, this needs to all be updated.

File: austation/code/modules/research/techweb/all_nodes.dm [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+/////////////////////////Mining tech/////////////////////////
+
+/datum/techweb_node/adv_mining
+	id = "adv_mining"
+	display_name = "Advanced Mining Technology"
+	description = "Efficiency Level 127"	//dumb mc references
+	prereq_ids = list("basic_mining", "adv_engi", "adv_power", "adv_plasma")
+	design_ids = list("drill_diamond", "jackhammer", "plasmacutter_adv", "mech_kinetic_accelerator", "hypermod", "hyperaoemod", "repeatermod", "resonatormod", "bountymod")
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
+	export_price = 5000
```
Comment (MCterra10) [inline austation/code/modules/research/techweb/all_nodes.dm:15]: what if bee changes this?

**Comment by Mick1299** [comment]
> what if bee changes this?
I'd suggest adding all of those mod techs to a new node.

**Comment by MrLostman** [comment]
mining loot list would need to be updated anyway each time bee adds something, because of the old cursed tank that is in there I could try to think a way to not just override it.

I'll add another node so it doesn't fuck with

File: austation/code/modules/research/techweb/all_nodes.dm (hunk 3 above)
Comment (MrLostman) [reply austation/code/modules/research/techweb/all_nodes.dm:15]: seems unlikely, but i'll add a new node

File: austation/code/modules/mining/necropolis_chests.dm (hunk 2 above)
Comment (MrLostman) [reply austation/code/modules/mining/necropolis_chests.dm:63]: I'll see if I can make some scuffed way to make it not just override it

**Review by MCterra10 — APPROVED** [review]

**Pull request merged**